modules/repository/repo.go

```diff
@@ -127,11 +127,20 @@ func MigrateRepositoryGitData(ctx context.Context, u *models.User, repo *models.
 	}
 
 	if opts.Mirror {
+		var interval = setting.Mirror.DefaultInterval
+		if opts.MirrorInterval != "" {
+			if interval, err = time.ParseDuration(opts.MirrorInterval); err != nil {
```

Comment:
Need to ensure that this is not too frequent:

```go
(interval != 0 && interval < setting.Mirror.MinInterval)
```
 take a look at routers/repo/setting.go:152

Reply:
Done. Split into two checks one for failing to convert an another for the number being smaller than the minimum.